src/realm/alloc_slab.cpp

```diff
@@ -945,6 +949,7 @@ ref_type SlabAlloc::attach_file(const std::string& file_path, Config& cfg)
         m_attach_mode = cfg.is_shared ? attach_SharedFile : attach_UnsharedFile;
     }
     catch (const DecryptionFailed&) {
+        note_reader_end(this);
```

Comment:
Looks strange as you have installed the scope exit thing. Could it not be done in the same way as in group_shared.cpp.